src/vs/workbench/contrib/files/browser/fileCommands.ts

```diff
@@ -639,19 +641,72 @@ KeybindingsRegistry.registerCommandAndKeybindingRule({
 						'languageId': {
 							'type': 'string'
 						},
-						'path': {
+					}
+				}
+			}
+		]
+	},
+	handler: async (accessor, args?: { languageId?: string; viewType?: string }) => {
+		const editorService = accessor.get(IEditorService);
+
+		await editorService.openEditor({
+			resource: undefined,
+			options: {
+				override: args?.viewType,
+				pinned: true
+			},
+			languageId: args?.languageId,
+		});
+	}
+});
+
+KeybindingsRegistry.registerCommandAndKeybindingRule({
+	weight: KeybindingWeight.WorkbenchContrib,
+	when: null,
+	id: NEW_FILE_COMMAND_ID,
+	description: {
+		description: NEW_FILE_LABEL,
+		args: [
+			{
+				isOptional: true,
+				name: 'New File args',
+				description: 'Shows a picker and creates a new file at the selected location',
+				schema: {
+					'type': 'object',
+					'properties': {
+						'viewType': {
+							'type': 'string'
+						},
+						'languageId': {
+							'type': 'string'
+						},
+						'fileName': {
 							'type': 'string'
 						}
 					}
 				}
 			}
 		]
 	},
-	handler: async (accessor, args?: { languageId?: string; viewType?: string; path?: string }) => {
+	handler: async (accessor, args?: { languageId?: string; viewType?: string; fileName?: string }) => {
 		const editorService = accessor.get(IEditorService);
+		const dialogService = accessor.get(IFileDialogService);
+		const pathService = accessor.get(IPathService);
+		const fileService = accessor.get(IFileService);
+
+		const createFileLocalized = nls.localize('newFileCommand.saveLabel', "Create File");
+		const defaultFileUri = joinPath(await dialogService.defaultFilePath(pathService.defaultUriScheme), args?.fileName ?? 'newfile.txt');
```

Comment:
Some feedback:
* why the explicit `pathService.defaultUriScheme`?
* should the default file name be `Untitled.txt` to align with untitled files?

Reply:
- It expected a scheme and I didn't know if it was safe to pass in file so I just passed in whatever the path service said the default was. Saw this in other places so thought it was the right thing to do.
- `Untitled.txt` makes sense

Reply:
I think when you do not specify a default scheme, the file dialog service will just take it, @alexr00 ?

Reply:
Removed schema filter, seems fine